<launch>
   <node name="joy_node" pkg="joy" type="joy_node">
     <param name="dev" value="/dev/input/js0"/>
  </node>
  <group ns="bebop_hand">
  <node pkg="hand_gesture_uav" type="node" name="hgu_node" output="screen" required="true"/>
  <node pkg="pid" type="controller" name="position_X">
    <remap from="setpoint" to="setpoint_pos_X"/>
    <remap from="state" to="state_pos_X"/>
    <remap from="control_effort" to="control_effort_pos_X"/>
    <param name="Kp" value="0.85" />
    <param name="Ki" value="0.0" />
    <param name="Kd" value="0.6" />
    <param name="upper_limit" value="0.85" />
    <param name="lower_limit" value="-0.85" />
    <param name="windup_limit" value="10" />
    <param name="cutoff_frequency" value="25" />
    <param name="max_loop_frequency" value="100.0" />
    <param name="min_loop_frequency" value="95.0" />
  </node>
  <node pkg="pid" type="controller" name="position_Y">
    <remap from="setpoint" to="setpoint_pos_Y"/>
    <remap from="state" to="state_pos_Y"/>
    <remap from="control_effort" to="control_effort_pos_Y"/>
    <param name="Kp" value="0.85" />
    <param name="Ki" value="0.0" />
    <param name="Kd" value="0.6" />
    <param name="upper_limit" value="0.85" />
    <param name="lower_limit" value="-0.85" />
    <param name="windup_limit" value="10" />
    <param name="cutoff_frequency" value="25" />
    <param name="max_loop_frequency" value="100.0" />
    <param name="min_loop_frequency" value="95.0" />
  </node>
  <node pkg="pid" type="controller" name="altittude">
    <remap from="setpoint" to="setpoint_alt"/>
    <remap from="state" to="state_alt"/>
    <remap from="control_effort" to="control_effort_alt"/>
    <param name="Kp" value="3.5" />
    <param name="Ki" value="0.0" />
    <param name="Kd" value="0.0001" />
    <param name="upper_limit" value="10" />
    <param name="lower_limit" value="-10" />
    <param name="windup_limit" value="10" />
    <param name="cutoff_frequency" value="25" />
    <param name="max_loop_frequency" value="100.0" />
    <param name="min_loop_frequency" value="95.0" />
  </node>
  <node pkg="pid" type="controller" name="heading">
    <remap from="setpoint" to="setpoint_hdg"/>
    <remap from="state" to="state_hdg"/>
    <remap from="control_effort" to="control_effort_hdg"/>
    <param name="Kp" value="0.02" />
    <param name="Ki" value="0.0" />
    <param name="Kd" value="0.005" />
    <param name="upper_limit" value="0.85" />
    <param name="lower_limit" value="-0.85" />
    <param name="windup_limit" value="10" />
    <param name="cutoff_frequency" value="25" />
    <param name="max_loop_frequency" value="100.0" />
    <param name="min_loop_frequency" value="95.0" />
  </node>
  <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" />
  </group>
</launch>
